wp-content/themes/haruoshi_origin/functions.php

```diff
@@ -189,3 +189,70 @@ function taxonomy_writer() {
 ));
 
 endif;
+
+/**
+ * ページネーション
+ */
+function pagination($pages = 1, $range = 1)
+{
+	$pages = (int)$pages; //float型で渡ってくるので明示的にint型 へ
+	$paged = get_query_var('paged') ? get_query_var('paged') : 1;
+
+	// 1ページしかない時
+	if ($pages === 1) {
+		return;
+	}
+
+	if(1 != $pages) { //全ページが１でない場合はページネーションを表示する
+		// 最初と最後のページにいるときは前後3ページを表示
+		$range = ($paged == 1 || $paged == $pages) ? 2 : 1;
+		
+		echo '<div class="pager">';
+		echo '<ul class="pagination">';
+		if($paged > 1) {
+			echo '<li class="pre"><a href="'.get_pagenum_link($paged - 1).'"><span class="arrow arrow-left"></span></a></li>';
+		}
+		if($paged > 2 && $pages > 3) {
+			echo '<li><a href="'.get_pagenum_link(1).'"><span class="page-numbers">1</span></a></li>';
+		}
+		if($paged > 3 && $pages > 4) {
+			echo '<li><span class="dot-line"></span></li>';
+		}
+		for ($i=1; $i <= $pages; $i++) {
+				if($i <= $paged + $range && $i >= $paged - $range){ // $paged ± $range 以内であればページ番号を出力
+					if ($paged == $i) {
+						echo '<li><a class="active" href="'.get_pagenum_link($i).'"><span class="page-numbers">'.$i.'</span></a></li>';
```

Comment:
activeなやつにはリンク要らない気がする。
あとクラスの付け方はこう。
```suggestion
						echo '<li><span class="pagination__page-numbers--active">'.$i.'</span></li>';
```